sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/AbstractStep.java

```diff
@@ -61,11 +61,14 @@ public Status getStatus() {
      * @param newStatus the new status to be set
      */
     protected void setStatus(Status newStatus) {
-        Status oldStatus = status;
-        status = newStatus;
-        logger.info("{}: changed status from: {} to: {} (interrupted={})",
-                getName(), oldStatus, newStatus, interrupted);
-
+        Status oldStatus;
+        synchronized (statusLock) {
+            oldStatus = status;
+            status = newStatus;
+            logger.info("{}: changed status from: {} to: {} (interrupted={})",
+                    getName(), oldStatus, newStatus, interrupted);
+        }
+        // Just in case, avoid possibility of deadlocks by calling out from outside the lock:
         if (!Objects.equals(oldStatus, newStatus)) {
```

Comment:
We're considering this `notifyObservers` safe outside of synchronization just because it will ultimately tell the observer the "true" latest status when it deals with the notification?